### PR TITLE
Use graphql for recent projects

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #3982

## Changes

* Use graphql to fetch recent projects

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

**Review by samshara — DISMISSED** [review]

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
ProjectItem is not getting appropriate props to render all the items it needs.
Selection of project is broken.

File: app/views/Home/index.tsx [hunk 1]
```diff
@@ -35,49 +36,60 @@ import RecentActivity from './RecentActivity';
...
+        description
+        startDate
+        endDate
+        analysisFramework {
+          id
+          title
+        }
+        createdBy {
+          displayName
+        }
```
Comment (tnagorra) [inline app/views/Home/index.tsx]: Always pull `id` even if it is not used.
This helps apollo to udpate the normalized cache

File: app/views/Home/index.tsx [hunk 2]
```diff
@@ -35,49 +36,60 @@ import RecentActivity from './RecentActivity';
...
+        leads {
+          totalCount
+        }
+        userGroups {
+          memberships {
+            role
+          }
```
Comment (tnagorra) [inline app/views/Home/index.tsx]: id is missing

File: app/views/Home/index.tsx [hunk 3]
```diff
@@ -35,49 +36,60 @@ import RecentActivity from './RecentActivity';
...
+        }
+        topTaggers {
+          count
+          name
+        }
```
Comment (tnagorra) [inline app/views/Home/index.tsx]: id is missing

File: app/views/Home/index.tsx [hunk 4]
```diff
@@ -35,49 +36,60 @@ import RecentActivity from './RecentActivity';
...
+          numberOfLeadsTaggedAndControlled
+          numberOfUsers
+          entriesActivity {
+            count
+            date
+          }
+          leadsActivity {
+            count
+            date
+          }
```
Comment (tnagorra) [inline app/views/Home/index.tsx]: Pull `id` if it is available

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**